backend/typescript/services/implementations/reviewService.ts

```diff
@@ -399,6 +480,266 @@ class ReviewService implements IReviewService {
 
     return result;
   }
+
+  async updateReviews(id: string, entity: ReviewRequestDTO): Promise<void> {
+    try {
+      await this.db.transaction(async (t) => {
+        const reviewToUpdate = await PgReview.findByPk(id, {
+          transaction: t,
+          include: [{ all: true, nested: true }],
+        });
+
+        if (!reviewToUpdate) {
+          throw new Error(`Review id ${id} not found`);
+        }
+
+        // delete tags
+        const newTagIds = entity.tags
+          .filter((tag: TagRequest) => typeof tag.id === "number")
+          .map((tag: TagRequest) => tag.id);
+        const allReviewTags = await PgReviewTag.findAll({
+          where: { review_id: id },
+          transaction: t,
+        });
+        const droppedTags = allReviewTags
+          .map((reviewTag: PgReviewTag) => reviewTag.tag_id)
+          .filter((oldId: number) => !newTagIds.includes(oldId));
+        await Promise.all(
+          droppedTags.map(async (tagId: number) => {
+            const dbTag = await PgTag.findByPk(tagId, { transaction: t });
+            if (!dbTag) {
+              throw new Error(`Tag id ${dbTag} not found`);
+            }
+            reviewToUpdate.$remove("tags", dbTag, { transaction: t });
+          }),
+        );
+
+        // update tags
+        await Promise.all(
+          entity.tags.map(async (tag: TagRequest) => {
+            if (tag.id) {
+              const tagToUpdate = await PgTag.findByPk(tag.id, {
+                transaction: t,
+              });
+              if (!tagToUpdate) {
+                throw new Error(`Tag id ${tag.id} not found`);
+              }
+              await tagToUpdate.update({ name: tag.name });
+            } else {
+              await this.findOrCreateTag(reviewToUpdate, tag, t);
+            }
+          }),
+        );
+
+        // remove books
```

Comment:
Why do we delete books from the book table but not authors? Should we keep the books as redundant data like authors for consistency?

Reply:
I don't see a lot of cases where an admin would want to delete a book and want to add it later. For the sake of simplicity and time, maybe we should leave it that way? @susheelk What do you think?

Reply:
We shouldn't be deleting authors, since an author could have written multiple books (ie one referenced by another review)